tests/tracer/test_writer.py

```diff
@@ -899,6 +931,19 @@ def test_flush_connection_reset(endpoint_test_reset_server, writer_class):
             writer.flush_queue(raise_exc=True)
 
 
+@pytest.mark.parametrize("writer_class", (AgentWriter, CIVisibilityWriter, NativeWriter))
+def test_flush_connection_incomplete_read(endpoint_test_incomplete_read_server, writer_class):
+    """Test that IncompleteRead errors are handled properly by resetting the connection"""
+    with override_env(dict(DD_API_KEY="foobar.baz")):
+        writer = writer_class("http://%s:%s" % (_HOST, _INCOMPLETE_READ_PORT))
```

Comment:
nit we can use f strings here. We should add a linting rule for this but honestly doesnt matter for tests.